cdap-ui/app/cdap/components/AbstractWidget/AbstractMultiRowWidget/AbstractRow.tsx

```diff
@@ -82,22 +87,35 @@ export default class AbstractRow<
   };
 
   public render() {
+    const { errors, value } = this.props;
+    let errorMsg = null;
+    if (errors && value) {
+      const errorObj = errors.find((error: IErrorObj) => error.element === value);
+      if (errorObj) {
+        errorMsg = errorObj.msg;
+      }
+    }
     return (
-      <div className={this.props.classes.root}>
-        {this.renderInput()}
+      <React.Fragment>
+        <div className={this.props.classes.root}>
+          {this.renderInput()}
 
-        <If condition={!this.props.disabled}>
-          <React.Fragment>
-            <IconButton onClick={this.props.addRow}>
-              <AddIcon fontSize="small" />
-            </IconButton>
+          <If condition={!this.props.disabled}>
+            <React.Fragment>
+              <IconButton onClick={this.props.addRow}>
+                <AddIcon fontSize="small" />
+              </IconButton>
 
-            <IconButton color="secondary" onClick={this.props.removeRow}>
-              <DeleteIcon fontSize="small" />
-            </IconButton>
-          </React.Fragment>
+              <IconButton color="secondary" onClick={this.props.removeRow}>
+                <DeleteIcon fontSize="small" />
+              </IconButton>
+            </React.Fragment>
+          </If>
+        </div>
+        <If condition={errorMsg}>
+          <div className={this.props.classes.errorText}>{errorMsg}</div>
```

Comment:
nit: This should its own component. We should just be getting/generating text and passing it on to the component. Would be useful with isolating the styling

Reply:
I will be creating a new component for this with more->less functionality. Just displaying the errors as is for now.